pkg/act/act.go

```diff
@@ -0,0 +1,295 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package act
+
+import (
+	"context"
+	"fmt"
+	"math"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/act"
+	"github.com/cilium/cilium/pkg/maps/ctmap"
+	"github.com/cilium/cilium/pkg/maps/lbmap"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/service"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	metricsUpdateInterval = 15 * time.Second
+	metricsTimeout        = 10 * time.Minute
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "act-metrics")
```

Comment:
For new code, please take a `slog.Logger` as a dependency and wire it through, it will automatically be module-scoped.